broadway/api/utils/multiqueue.py

```diff
@@ -0,0 +1,45 @@
+from queue import Queue, Empty
+
+"""
+A wrapper around Python's queue module that contains multiple queues
+and round-robins between them when pulling.
+"""
+
+
+class MultiQueue:
+    def __init__(self):
+        self.queues = {}
+        self.keys = []
+        self.round_robin_idx = 0
+
+    def add_queue(self, queue_id):
+        if queue_id in self.queues:
+            raise Exception(f"{queue_id} already exists in the MultiQueue.")
+
+        self.queues[queue_id] = Queue()
+        self.keys.append(queue_id)
+
+    def push(self, queue_id, elem):
+        if queue_id not in self.queues:
+            raise Exception(f"{queue_id} does not exist in the MultiQueue.")
```

Comment:
Could we not use string interpolation here? I don't think other parts of the API use it and it creates a dependency on python 3.6

Reply:
Wait, Broadway is already tied to Python 3.6. That is the minimum supported version.

F-strings are better imo (more readable), I would suggest that we keep f strings.

Reply:
Switching back to f-strings